sklearnex/covariance/incremental_covariance.py

```diff
@@ -117,14 +232,71 @@ def fit(self, X, queue=None):
         self : object
             Returns the instance itself.
         """
-        n_samples, n_features = X.shape
-        if self.batch_size is None:
-            batch_size_ = 5 * n_features
+
+        dispatch(
+            self,
+            "fit",
+            {
+                "onedal": self.__class__._onedal_fit,
+                "sklearn": None,
+            },
+            X,
+        )
+        return self
+
+    def _onedal_fit(self, X, queue=None):
+        self.n_samples_seen_ = 0
+
+        if sklearn_check_version("1.2"):
```

Comment:
does it makes sense to move such stuff to fit method for consistency with other _onedal_* methods?

Reply:
if check_array is used before dispatch, it could strip away the queue in cases of dpnp and dpctl.  Since we don't test fallbacks to CPU yet, this could be silently masking issues with the underlying SYCL support.

Reply:
but you use check_array before dispatch in partial_fit. moreover, it is used like that in sklearnex.EmpiricalCovariance. it looks like it is my bad because i was not able to find such usage in other estimators. probably should be fixed everywhere then. it might be worth to avoid check_array usage in sklearnex part and move it to onedal package.

Reply:
but i don't really like adding scikit dependency to onedal package

Reply:
Maybe I misunderstand, this is in the sklearnex/covariance section and not in the onedal package.